src/classes/HMSSDK.tsx

```diff
@@ -1105,10 +1108,10 @@ export class HMSSDK {
    * @param {*} callback
    * @memberof HMSSDK
    */
-  addEventListener = (
-    action: HMSUpdateListenerActions | HMSPIPListenerActions,
-    callback: any
-  ) => {
+  addEventListener<T extends HMSUpdateListenerActions>(
+    action: T | HMSPIPListenerActions,
+    callback: HMSUpdateListenerCallbacks[T]
+  ) {
```

Comment:
change needed to keep `addEventListener` an arrow function

```suggestion
  ) => {
```